src/generate_bristol.rs

```diff
@@ -129,11 +129,11 @@ fn collect_inputs(
     visited: &mut HashSet<usize>,
     bool: &BoolWire,
 ) {
-    if let Some(id) = bool.id() {
-        if !visited.insert(id) {
-            return;
-        }
-    } else {
+    let Some(id) = bool.id() else {
+        return;
+    };
+
+    if !visited.insert(id) {
```

Comment:
Unrelated to the stated purpose of the PR. Just noticed an opportunity to simplify using the newer syntax. (See [let-else](https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html).)